frontend/awx/common/PageFormLabelSelect.tsx

```diff
@@ -0,0 +1,63 @@
+import { ReactElement, useEffect, useState } from 'react';
+import { FieldPath, FieldValues } from 'react-hook-form';
+import { PageFormCreatableSelect } from '../../../framework/PageForm/Inputs/PageFormCreatableSelect';
+import { useTranslation } from 'react-i18next';
+import { Label } from '../interfaces/Label';
+import { ItemsResponse, requestGet } from '../../common/crud/Data';
+
+/**
+ * Component to select and/or create labels in a form
+ */
+export function PageFormLabelSelect<
+  TFieldValues extends FieldValues = FieldValues,
+  TFieldName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
+>(props: {
+  labelHelpTitle: string;
+  labelHelp: string;
+  name: TFieldName;
+  placeholderText?: string;
+  additionalControls?: ReactElement;
+}) {
+  const { labelHelpTitle, labelHelp, name, placeholderText, additionalControls } = props;
+  const { t } = useTranslation();
+  const [labelOptions, setLabels] = useState<Label[]>();
+
+  useEffect(() => {
+    async function fetchLabels() {
+      const allLabels = [];
+      try {
+        const labels = await requestGet<ItemsResponse<Label>>(
+          `/api/v2/labels/?order_by=name&page=1&page_size=200`
+        );
+        allLabels.push(...labels.results);
```

Comment:
Suggestion: add a check here to make sure duplicate label names aren't being added.

Reply:
@kialam Good catch! This requires enhancements to the underlying framework component (PageFormCreatableSelect). I also noticed that the existing implementation (also used in the Template form) does not include the option to create a new label. I've created a new Jira issue for both enhancements and would prefer to address them separately.